patroni/dcs/consul.py

```diff
@@ -472,6 +476,7 @@ def _update_service(self, data):
             'check': check,
             'tags': tags,
             'enable_tag_override': True,
+            'token': self._token,
```

Comment:
https://developer.hashicorp.com/consul/api-docs/api-structure#authentication
> Previously this was provided via a `?token=` query parameter. This functionality exists on many endpoints for backwards compatibility, but its use is highly discouraged, since it can show up in access logs as part of the URL.

Patroni is already using `X-Consul-Token` header for all HTTP requests: https://github.com/zalando/patroni/blob/ef2922fe37206ac376abb87d53b5a01d4e3b57c8/patroni/dcs/consul.py#L120-L122
E.g., there is no need to explicity add the token parameter. We only need to call register_service() if the token has changed. Also it will make https://github.com/zalando/patroni/pull/2450/files#diff-99372e752231ac1fddf9cda076dd985f0ed8293e8ba1c06ea1d95a0530ccb959R446 unnecessary.

Reply:
Ah, good point.  Ok, I think I've fixed it up the way you are suggesting.  Let me know if I've missed anything.  Thanks.